db/migrations/00030_hierarchical_groups.sql

```diff
@@ -0,0 +1,22 @@
+-- +goose Up
+-- +goose StatementBegin
+CREATE TABLE group_hierarchies (
+    id UUID PRIMARY KEY NOT NULL DEFAULT gen_random_uuid(),
+    parent_group_id UUID NOT NULL REFERENCES groups(id),
+    member_group_id UUID NOT NULL REFERENCES groups(id),
```

Comment:
do we want to prevent duplicate parent -> member entries?
should we prevent a parent being part of the member here?

Reply:
Like a `parent_group_id <> member_group_id` constraint? Yeah happy to add that. It should also be caught before that point by the cycle detection that I'm unfortunately going to have to implement in Go though.

Reply:
yes that and maybe a a unique parent_group_id + member_group_id, however that alone wouldn't stop everything. and I can't find a good way to do the equivalent of an `EXCLUDE using @>` style DDL in crdb so it might not be seen as worth it. as a lot of the heavy lifting for integrity will need to be in GO still.